stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -92,6 +92,60 @@ isposdef(A::AbstractMatrix) =
     ishermitian(A) && isposdef(cholesky(Hermitian(A); check = false))
 isposdef(x::Number) = imag(x)==0 && real(x) > 0
 
+"""
+ispossemdef(A, k) -> Bool
+
+Test whether a matrix is positive semi-definite with specified rank `k` by
+checking that `k` of its eigenvalues are positive and the rest are zero.
+
+# Examples
+```jldoctest
+julia> A = [1 0; 0 0]
+2×2 Array{Int64,2}:
+ 1  0
+ 0  0
+
+julia> ispossemdef(A, 1)
+true
+
+julia> ispossemdef(A, 2)
+false
+"""
+function ispossemdef(X::AbstractMatrix, k::Int;
+                     atol::Real = 0.0,
```

Comment:
```suggestion
                     atol::Real=0.0,
```
and elsewhere

Reply:
Looks like you only changed this occurrence, but the space should be removed in all of the added code with keyword arguments.